sympy/functions/special/hyper.py

```diff
@@ -288,7 +288,58 @@ def _eval_nseries(self, x, n, logx, cdir=0):
             den = Mul(*[RisingFactorial(b, i) for b in bq])
             terms.append(((num/den) * (arg**i)) / factorial(i))
 
-        return (Add(*terms) + Order(x**n,x))
+        result = Add(*terms)
+        newn = min((Abs(a) for a in ap if a.is_negative and a.is_integer), default=n)
+
+        if newn < n:
+            return result
+        return result + Order(x**n, x)
+
+    def _eval_aseries(self, n, args0, x, logx):
+
+        # The asymptotic expansion of the hypergeometric functions is quite complicated
+        # and are not well defined for all relation between the parameters p and q.
+        # The following is a simple implementation of the asymptotic expansion of the
+        # hypergeometric function for the case when p = q + 1 and |z| -> oo.
```

Comment:
I also want to see a citation for these formulae.

Reply:
Specifically what I want to know is what are the limitations on the validity of the formulae.

Reply:
+1 
It would be great if this info can be backed by some reference/citation.

Reply:
Thanks for reviewing. I think, maybe the following [dmlf](https://dlmf.nist.gov/16.11#E6) page can be used as a reference here, for reference one can compare the results of this page to a more simplified form of the same expression given in the wolfram functions website quoted in the comments above. Also please correct me if I am misunderstanding by what is meant by an "acceptable" citation here as the very same formula is present in the wolfram functions website.